src/20240209_AaveV3Ethereum_TreasuryManagementGSMFundingRWAStrategyPreparationsPart2/AaveV3Ethereum_TreasuryManagementGSMFundingRWAStrategyPreparationsPart2_20240209.sol

```diff
@@ -0,0 +1,146 @@
+ * - Discussion: https://governance.aave.com/t/arfc-orbit-program-renewal/16550
+ */
+contract AaveV3Ethereum_TreasuryManagementGSMFundingRWAStrategyPreparationsPart2_20240209 is
+  DepositV3SwapPayload
+{
+  using SafeERC20 for IERC20;
+
+  uint256 public constant USDC_TO_SWAP = 700_000e6;
+  AaveSwapper public constant SWAPPER = AaveSwapper(MiscEthereum.AAVE_SWAPPER);
+  address public constant MILKMAN = 0x11C76AD590ABDFFCD980afEC9ad951B160F02797;
+  address public constant PRICE_CHECKER = 0xe80a1C615F75AFF7Ed8F08c9F21f9d00982D666c;
+
+  function execute() external {
+    _swap();
+    _orbit();
+  }
+
+  /**
+   * @dev To be used after the async swap takes place to deposit USDT in AAVE v3
+   */
+  function deposit(address token, uint256 amount) external {
+    _deposit(token, amount);
+  }
+
+  function _swap() internal {
+    // Transfer 700k USDC from Collector to Swapper
+    AaveV3Ethereum.COLLECTOR.transfer(
+      AaveV3EthereumAssets.USDC_UNDERLYING,
+      address(SWAPPER),
+      USDC_TO_SWAP
+    );
+
+    // Deposit all remaining USDC from collector into Aave v3.
+    uint256 usdcBalance = IERC20(AaveV3EthereumAssets.USDC_UNDERLYING).balanceOf(
+      address(AaveV3Ethereum.COLLECTOR)
+    );
+
+    AaveV3Ethereum.COLLECTOR.transfer(
+      AaveV3EthereumAssets.USDC_UNDERLYING,
+      address(this),
+      usdcBalance
+    );
+
+    _deposit(AaveV3EthereumAssets.USDC_UNDERLYING, usdcBalance);
+
+    // Transfer the whole DAI balance of the collector into the Swapper
+    uint256 daiBalance = IERC20(AaveV3EthereumAssets.DAI_UNDERLYING).balanceOf(
+      address(AaveV3Ethereum.COLLECTOR)
+    );
+
+    AaveV3Ethereum.COLLECTOR.transfer(
+      AaveV3EthereumAssets.DAI_UNDERLYING,
+      address(SWAPPER),
+      daiBalance
+    );
+
+    // Swap DAI into USDT with receiver being this contract
+    SWAPPER.swap(
+      MILKMAN,
+      PRICE_CHECKER,
+      AaveV3EthereumAssets.DAI_UNDERLYING,
+      AaveV3EthereumAssets.USDT_UNDERLYING,
+      AaveV3EthereumAssets.DAI_ORACLE,
+      AaveV3EthereumAssets.USDT_ORACLE,
+      SELF,
+      daiBalance,
+      50
+    );
+
+    // Swap USDC into USDT with receiver being this contract
+    SWAPPER.swap(
+      MILKMAN,
+      PRICE_CHECKER,
+      AaveV3EthereumAssets.USDC_UNDERLYING,
+      AaveV3EthereumAssets.USDT_UNDERLYING,
+      AaveV3EthereumAssets.USDC_ORACLE,
+      AaveV3EthereumAssets.USDT_ORACLE,
+      SELF,
+      USDC_TO_SWAP,
+      50
+    );
+  }
+
+  function _orbit() internal {
+    AaveV3Ethereum.COLLECTOR.transfer(
+      AaveV3EthereumAssets.WETH_UNDERLYING,
+      address(this),
+      OrbitProgramData.TOTAL_WETH_REBATE
+    );
+
+    IWETH(AaveV3EthereumAssets.WETH_UNDERLYING).withdraw(OrbitProgramData.TOTAL_WETH_REBATE);
+
+    OrbitProgramData.GasUsage[] memory usage = OrbitProgramData.getGasUsageData();
+    uint256 usageLength = usage.length;
+    for (uint256 i = 0; i < usageLength; i++) {
```

Comment:
provably better to use directly usage.length, as usageLength is not reused

Reply:
it's used to save gas